tests/test_api_build.py

```diff
@@ -1844,8 +1850,19 @@ def test_warning_on_file_clobbering(testing_config, capfd):
         config=testing_config,
     )
     # The clobber warning here is raised when creating the test environment for b
-    out, err = capfd.readouterr()
-    assert "ClobberWarning" in err
+    if Version(conda_version) >= Version("24.9.0"):
+        # conda >=24.9.0
+        clobber_warning_found = False
+        for record in caplog.records:
+            if "ClobberWarning:" in record.message:
+                clobber_warning_found = True
+        assert clobber_warning_found
+    else:
+        # before the new lazy index added in conda 24.9.0
+        # see https://github.com/conda/conda/commit/1984b287548a1a526e8258802a6f1fec2a11ecc3
+        out, err = capfd.readouterr()
+        assert "ClobberWarning" in err
+
```

Comment:
This is a fix from https://github.com/conda/conda-build/pull/5488 that we need to get this merged